app/controllers/hours_controller.rb

```diff
@@ -0,0 +1,71 @@
+class HoursController < ApplicationController
+  include MakeNotice
+  before_action :set_hour, only: [:show, :edit, :update, :destroy]
+
+  def index
+    authorize Hour
+    @hours = Hour.all
+  end
+
+  def show; end
+
+  def new
+    @hour = Hour.new(volunteer_id: params[:volunteer_id])
+    @assignments_clients = select_clients
+    authorize @hour
+  end
+
+  def edit; end
+
+  def create
+    @hour = Hour.new(hour_params)
+    authorize @hour
+    if @hour.save
+      redirect_to @hour, make_notice
+    else
+      render :new
+    end
+  end
+
+  def update
+    if @hour.update(hour_params)
+      redirect_to @hour, make_notice
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @hour.destroy
+    if current_user.superadmin?
+      redirect_to hours_url, make_notice
+    else
+      redirect_to volunteer_hours_volunteer_path(@hour.volunteer), make_notice
+    end
+  end
+
+  private
+
+  def select_clients
```

Comment:
I like that much better 👍 
One thing comes to my mind tho:

How about making the class variable directly in that select_clients method and call this by `after_action :select_clients, only: [:new]`?

This way the action can be:
```ruby
after_action :select_client, only: :new

# ...

def new
  @hour = Hour.new(volunteer_id: params[:volunteer_id])
  authorize @hour
end

# ...

def select_client
   # ...
     @assignments_clients = ...
```

Reply:
You can also use the Rails helper `options_from_collection_for_select(@assignments_clients, :id, :to_s)` in the view instead of doing the `.map` here, and then I would just put `@assignments_clients = Assignment.where(volunteer: params[:volunteer_id])` directly into the `#new` method.